### DISCUSSION TITLE
How to spawn a not static thread?

### DISCUSSION THREAD

**Discussion by lz1998** [tokio-rs/tokio#4218] [Q&A]
I ensure the thread will be finished before the method is finished with `JoinHandle`.
```rust

impl X{
    pub fn execute_multi(&self){
        let mut handles =Vec::new();
        for _ in 1..20{
            handles.push(tokio::spawn(async move{
                self.execute();
            }));
        }
        for h in handles.iter_mut(){
            h.join();
        }
    }
}

```

**Comment by Darksonn** [comment]
You appear to have mixed up the concepts of "thread" and "task". A thread is what you get when you use `std::thread::spawn`, and a task is what you get when you use `tokio::spawn`. Only threads have a `.join()` method, as the task `JoinHandle` is instead directly `.await`ed.

If your question is about tasks, then you can find the answer [here](https://stackoverflow.com/questions/65269738/spawn-non-static-future-with-tokio/65287449#65287449).
